core/food_apis/scheduler.py

```
@@ -66,7 +66,13 @@ def _setup_signal_handlers(self):
 
         def signal_handler(signum, frame):
             logger.info(f"Received signal {signum}, initiating graceful shutdown...")
-            asyncio.create_task(self.stop())
+            # In some test/client contexts there may be no running loop (sync thread)
+            try:
+                loop = asyncio.get_running_loop()
+                loop.create_task(self.stop())
```

Ruff (0.14.0)

Store a reference to the return value of `loop.create_task`

(RUF006)